deltaspike/src/main/java/cucumber/runtime/java/deltaspike/DeltaSpikeObjectFactory.java

```diff
@@ -0,0 +1,36 @@
+package cucumber.runtime.java.deltaspike;
+
+import cucumber.api.java.ObjectFactory;
+import javax.enterprise.inject.spi.CDI;
+import org.apache.deltaspike.cdise.api.CdiContainer;
+import org.apache.deltaspike.cdise.api.CdiContainerLoader;
+
+public class DeltaSpikeObjectFactory implements ObjectFactory {
+
+    private final CdiContainer container;
+
+    public DeltaSpikeObjectFactory() {
+        this.container = CdiContainerLoader.getCdiContainer();
+    }
+
+    @Override
+    public void start() {
+        container.boot();
+    }
+
+    @Override
+    public void stop() {
+        container.shutdown();
+    }
+
+    @Override
+    public boolean addClass(final Class<?> clazz) {
+        return true;
+    }
+
+    @Override
+    public <T> T getInstance(final Class<T> type) {
+        return CDI.current().select(type).get();
```

Comment:
this is the "default" way to get a type for a CDI-Container? The `container` is a deltaspike interface to start and stop containers in java-se it does not contains a easy to use method to get the requested object.

Reply:
I wasn't aware. Cheers.

Reply:
CDI comes from cdi 2 where deltaspike usage can be replace by SeContainer. Did you intend to use BeanManager to do the lookup or deltaspike BeanProvider directly?

Reply:
CDI comes from CDI 1.1? SeContainer with 2.0. Why should I use BeanManager at this place?
It should be possible to extend addClass to register given type to the container, but think CDI default discovery should be good enough her?

Reply:
Well there are two things

1. As you mentionned cdi comes from spec v1.1 (not 2 as i mentionned) but point is deltaspike api is really portable for cdi >= 1.0 containers so you cant use it together
2. Yes scanning is a good default but even in this case you dont need cdi class since deltaspike has a replacement for it

In other words it looks a hybrid cdi 2 (secontainer+cdi) / deltaspike (cdictrl/beanprovider) impl to me so can be worth cleaning up IMHO.

Does it make sense presented this way?

Indeed i prefer the option with v2 but i agree a v1.0 still makes sense today.

Reply:
2 - i do not understand.

to solve the hybrid you prefer beanprovider over CDI?

Reply:
Yes please

Point 2 was just that CDI is the standard of BeanProvider but nit available for cdictrl based on cdi 1.0 impls.

Side note you can need to handle creational contexts to release them to not leak.